R/project.R

```diff
@@ -146,7 +202,33 @@ Project <-
         private$.simulationResults <- allSimulationResults[names(self$activeScenarios)]
 
         invisible(self)
+      },
+      #' @description Validate the project configuration.
+      #' This method will check if all references in scenarios are valid and print warnings if any.
+      status = function() {
+
+        purrr::walk(self$configurations$scenarios, ~ .x$validate())
+        warnings <- private$.warningManager$get_warnings()
+
+        if (length(warnings) > 0) {
+          cli::cli_alert_info("Warnings:")
+          cli::cli_ul()
+          for (scenario_name in names(warnings)) {
+            for (code in names(warnings[[scenario_name]])) {
+              cli::cli_alert_warning(paste0("Scenario: ", scenario_name, " - ", warnings[[scenario_name]][[code]]))
+            }
+          }
+          cli::cli_end()
+        } else {
+          cli::cli_alert_success("No warnings found.")
```

Comment:
Add argument to status so that:
- when status() is explicitly called this is printed,
- when configuration is modified, this is **not** printed.

Reply:
For example:
```r
p$configurations$scenarios$TestScenario$individual <- "Indiv1"

p$status()
# ✔ No warnings found.
```